client/src/app/recipe-search/recipe-search.component.ts

```diff
@@ -9,28 +14,74 @@ import { distinctUntilChanged } from 'rxjs/operators';
   styleUrl: './recipe-search.component.scss',
 })
 export class RecipeSearchComponent {
-  public searchedQuery = '';
-  public inputText = '';
-  public isLoading: boolean = false;
   public recipeList$: Observable<Recipe[]>;
-  private inputText$: Subject<string> = new Subject();
+  public inputText: string = '';
+  public isLoading: boolean = false;
+  public isFilterAndSortOpen: boolean = false;
+  public isFilterAndSortVisible: boolean = false;
+  public searchModel: SearchModel = {
+    searchQuery: '',
+    filterSortOptions: {
+      filterBy: '',
+      filterString: '',
+      sortBy: '',
+      isReversed: false,
+    },
+  };
+
+  private searchModel$: Subject<SearchModel> = new Subject();
 
   constructor(private recipeService: RecipeService) {
-    this.recipeList$ = this.inputText$.pipe(
+    this.recipeList$ = this.searchModel$.pipe(
       distinctUntilChanged(
-        (prev, curr) => prev.toLowerCase() === curr.toLowerCase()
+        (prev, curr) =>
+          prev.searchQuery.toLowerCase() === curr.searchQuery.toLowerCase() &&
```

Comment:
This needs cleanup. Compare the two properties of searchModel.

Reply:
Tried that but doesn't work because the second property of searchModel is an object which apparently you can't check with === or ==. Only works if I check all the values of the object directly.